### DISCUSSION TITLE
Missing key(s) in state_dict

### DISCUSSION THREAD

**Discussion by purvang3** [rwightman/efficientdet-pytorch#271] [Q&A] [answered]
Hi @rwightman , Thank you for your amazing work.

I having difficulty evaluating trained model.

I trained model for 100 epoch and wanted to evaluate trained model using validate.py, but I am getting following error.

"""
RuntimeError: Error(s) in loading state_dict for EfficientDet:
        Missing key(s) in state_dict: "fpn.resample.3.conv.conv.bias", "fpn.cell.0.fnode.0.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.1.combine.resample.2.conv.conv.bias", "fpn.cell.0.fnode.1.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.2.combine.resample.1.conv.conv.bias", "fpn.cell.0.fnode.2.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.3.combine.resample.0.conv.conv.bias", "fpn.cell.0.fnode.3.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.4.combine.resample.1.conv.conv.bias", "fpn.cell.0.fnode.4.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.5.combine.resample.2.conv.conv.bias", "fpn.cell.0.fnode.5.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.6.after_combine.conv.conv_pw.bias", "fpn.cell.0.fnode.7.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.0.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.1.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.2.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.3.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.4.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.5.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.6.after_combine.conv.conv_pw.bias", "fpn.cell.1.fnode.7.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.0.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.1.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.2.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.3.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.4.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.5.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.6.after_combine.conv.conv_pw.bias", "fpn.cell.2.fnode.7.after_combine.conv.conv_pw.bias", "class_net.conv_rep.0.conv_pw.bias", "class_net.conv_rep.1.conv_pw.bias", "class_net.conv_rep.2.conv_pw.bias", "box_net.conv_rep.0.conv_pw.bias", "box_net.conv_rep.1.conv_pw.bias", "box_net.conv_rep.2.conv_pw.bias". 
"""

I trained model using following command.

./distributed_train.sh 2 ~/models/datasets/coco/ --model efficientdet_d0 -b 12 --amp --lr 0.009 --sync-bn --warmup-epochs 5 --lr-noise 0.4 0.9 --model-ema --model-ema-decay 0.9999 --smoothing 0.05 

Evaluation using following command.

python validate.py ~/models/datasets/coco/ --model tf_efficientdet_d0 --checkpoint output/train/20220315-062951-efficientdet_d0/checkpoint-83.pth.tar --use-ema

Thank you.

**Comment by rwightman** [comment, marked as answer]
@purvang3 you are training and validating with different models, `tf_` prefix has extra bias layers and different padding.... remove the tf_ from the model name for validation
